src/TemplateProcessor.ts

```diff
@@ -1136,7 +1157,7 @@ export default class TemplateProcessor {
                 "materialized": true
             }
         );
-        await this.callDataChangeCallbacks(data, jsonPtr);
+        await this.callDataChangeCallbacks(data, jsonPtr, op === 'delete');
```

Comment:
wait, you are claiming op is "delete" in every case? But we can clearly see we are allowed to 'set' data into an untracked location. So how can it always be 'delete'. Can't be right.